server/src/main/java/org/apache/druid/indexing/overlord/supervisor/Supervisor.java

```diff
@@ -86,6 +87,11 @@ default Boolean isHealthy()
     return null; // default implementation for interface compatability; returning null since true or false is misleading
   }
 
+  default SupervisorTaskAutoScaler createAutoscaler()
+  {
+    return null;
+  }
```

Comment:
It seems a little untidy but the default impl should do the same thing that the existing impl does, so that we do not break extensions that use auto-scalers.

```suggestion
  default SupervisorTaskAutoScaler createAutoscaler(SupervisorSpec spec)
  {
    return spec.createAutoscaler(this);
  }
```